? .deps
? .pch
? bk-deps
? bk-make-pch
? patch_pnghandler.diff
? patch_pnghandler_abort.diff
? version-script
? contrib/Makefile
? contrib/samples/Makefile
? contrib/samples/animate/Makefile
? contrib/samples/applet/Makefile
? contrib/samples/deprecated/Makefile
? contrib/samples/deprecated/proplist/Makefile
? contrib/samples/deprecated/resource/Makefile
? contrib/samples/deprecated/treelay/Makefile
? contrib/samples/fl/Makefile
? contrib/samples/foldbar/extended/Makefile
? contrib/samples/foldbar/foldpanelbar/Makefile
? contrib/samples/gizmos/Makefile
? contrib/samples/gizmos/dynsash/Makefile
? contrib/samples/gizmos/dynsash_switch/Makefile
? contrib/samples/gizmos/editlbox/Makefile
? contrib/samples/gizmos/led/Makefile
? contrib/samples/gizmos/multicell/Makefile
? contrib/samples/gizmos/splittree/Makefile
? contrib/samples/mmedia/Makefile
? contrib/samples/ogl/Makefile
? contrib/samples/ogl/ogledit/Makefile
? contrib/samples/ogl/studio/Makefile
? contrib/samples/plot/Makefile
? contrib/samples/stc/Makefile
? contrib/samples/svg/Makefile
? contrib/src/Makefile
? contrib/src/animate/Makefile
? contrib/src/applet/Makefile
? contrib/src/deprecated/Makefile
? contrib/src/fl/Makefile
? contrib/src/foldbar/Makefile
? contrib/src/gizmos/Makefile
? contrib/src/mmedia/Makefile
? contrib/src/net/Makefile
? contrib/src/ogl/Makefile
? contrib/src/plot/Makefile
? contrib/src/stc/Makefile
? contrib/src/svg/Makefile
? contrib/utils/Makefile
? contrib/utils/convertrc/Makefile
? demos/Makefile
? demos/bombs/Makefile
? demos/dbbrowse/Makefile
? demos/forty/Makefile
? demos/fractal/Makefile
? demos/life/Makefile
? demos/poem/Makefile
? lib/libwx_based-2.6.so.0.3.0
? lib/libwx_based_net-2.6.so.0.3.0
? lib/libwx_based_xml-2.6.so.0.3.0
? lib/libwx_gtk2d_adv-2.6.so.0.3.0
? lib/libwx_gtk2d_core-2.6.so.0.3.0
? lib/libwx_gtk2d_html-2.6.so.0.3.0
? lib/libwx_gtk2d_media-2.6.so.0.3.0
? lib/libwx_gtk2d_qa-2.6.so.0.3.0
? lib/libwx_gtk2d_xrc-2.6.so.0.3.0
? lib/wx
? samples/Makefile
? samples/artprov/Makefile
? samples/calendar/Makefile
? samples/caret/Makefile
? samples/checklst/Makefile
? samples/config/Makefile
? samples/console/Makefile
? samples/controls/Makefile
? samples/debugrpt/Makefile
? samples/dialogs/Makefile
? samples/dialup/Makefile
? samples/display/Makefile
? samples/dnd/Makefile
? samples/docview/Makefile
? samples/docvwmdi/Makefile
? samples/dragimag/Makefile
? samples/drawing/Makefile
? samples/dynamic/Makefile
? samples/erase/Makefile
? samples/event/Makefile
? samples/except/Makefile
? samples/exec/Makefile
? samples/font/Makefile
? samples/grid/Makefile
? samples/help/Makefile
? samples/htlbox/Makefile
? samples/html/Makefile
? samples/html/about/Makefile
? samples/html/help/Makefile
? samples/html/helpview/Makefile
? samples/html/printing/Makefile
? samples/html/test/Makefile
? samples/html/virtual/Makefile
? samples/html/widget/Makefile
? samples/html/zip/Makefile
? samples/image/Makefile
? samples/internat/Makefile
? samples/ipc/Makefile
? samples/joytest/Makefile
? samples/keyboard/Makefile
? samples/layout/Makefile
? samples/listbox/Makefile
? samples/listctrl/Makefile
? samples/mdi/Makefile
? samples/menu/Makefile
? samples/minifram/Makefile
? samples/minimal/Makefile
? samples/mobile/Makefile
? samples/mobile/styles/Makefile
? samples/mobile/wxedit/Makefile
? samples/multimon/Makefile
? samples/notebook/Makefile
? samples/png/Makefile
? samples/popup/Makefile
? samples/printing/Makefile
? samples/propsize/Makefile
? samples/render/Makefile
? samples/richedit/Makefile
? samples/rotate/Makefile
? samples/sashtest/Makefile
? samples/scroll/Makefile
? samples/scrollsub/Makefile
? samples/shaped/Makefile
? samples/sockets/Makefile
? samples/sound/Makefile
? samples/splash/Makefile
? samples/splitter/Makefile
? samples/statbar/Makefile
? samples/taskbar/Makefile
? samples/text/Makefile
? samples/thread/Makefile
? samples/toolbar/Makefile
? samples/treectrl/Makefile
? samples/typetest/Makefile
? samples/validate/Makefile
? samples/vscroll/Makefile
? samples/widgets/Makefile
? samples/wizard/Makefile
? samples/xrc/Makefile
? utils/Makefile
? utils/HelpGen/Makefile
? utils/HelpGen/src/Makefile
? utils/configtool/Makefile
? utils/configtool/src/Makefile
? utils/emulator/Makefile
? utils/emulator/src/Makefile
? utils/helpview/Makefile
? utils/helpview/src/Makefile
? utils/hhp2cached/Makefile
? utils/tex2rtf/Makefile
? utils/tex2rtf/src/Makefile
? utils/wxrc/.deps
? utils/wxrc/Makefile
? utils/wxrc/wxrc
Index: src/common/imagpng.cpp
===================================================================
RCS file: /pack/cvsroots/wxwidgets/wxWidgets/src/common/imagpng.cpp,v
retrieving revision 1.56
diff -b -u -2 -r1.56 imagpng.cpp
--- src/common/imagpng.cpp	2005/07/26 09:05:23	1.56
+++ src/common/imagpng.cpp	2006/01/21 19:25:13
@@ -175,8 +175,14 @@
 // from pngerror.c
 // so that the libpng doesn't send anything on stderr
+void // prototype, used by wx_png_error
+PNGLINKAGEMODE wx_png_warning(png_structp png_ptr, png_const_charp message);
+
 void
 PNGLINKAGEMODE wx_png_error(png_structp WXUNUSED(png_ptr), png_const_charp message)
 {
-    wxLogFatalError( wxString::FromAscii(message) );
+	// JS: deliver it to wx_png_warning and don't do any more actions
+	// libpng will jump back to the calling function (LoadFile and SaveFile) and
+	// they will handle this error
+    wx_png_warning(NULL, message);
 }
 

 	  	 
